### PR TITLE
CAPV: Release v29.0.0.

### PR DESCRIPTION
**Pull request by njuettner**
Towards: https://github.com/giantswarm/roadmap/issues/3710

### Checklist

- [x] Roadmap issue created
- [x] Release uses latest stable Flatcar
- [x] Release uses latest Kubernetes patch version

### Triggering E2E tests

To trigger the E2E test for each new Release added in this PR, add a comment with the following:

`/run releases-test-suites`

If you want to trigger conformance tests, you can do so by adding a comment similar to the following:

`/run conformance-tests PROVIDER=capa RELEASE_VERSION=29.1.0`

For more details see the [README.md](/README.md#running-tests-against-prs).

### CODE REVIEW

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard

**Comment by Gacko** [comment]
Merge at your own discretion! 🙂

**Review by Gacko — APPROVED** [review]

File: vsphere/v29.0.0/release.yaml
Comment (vxav) [inline vsphere/v29.0.0/release.yaml:98]:
```suggestion
    version: 3975.2.2
```

File: vsphere/v29.0.0/release.diff
Comment (vxav) [inline vsphere/v29.0.0/release.diff:98]:
```suggestion
    version: 3815.2.5						|           version: 3975.2.2
```

File: vsphere/v29.0.0/README.md
Comment (vxav) [inline vsphere/v29.0.0/README.md:7]:
```suggestion
- Flatcar from 3815.2.5 to [3975.2.2](https://www.flatcar.org/releases#release-3975.2.2)
```

File: vsphere/v29.0.0/release.yaml
Comment (Gacko) [reply vsphere/v29.0.0/release.yaml:98]: I also came across this and talked to Nick. We agreed on going with `3975.2.1` for now as this is the same version we're currently using in the other providers.

Let's use `3975.2.1` for v29.0.0 and bump in future releases across all providers.

File: vsphere/v29.0.0/README.md
Comment (Gacko) [reply vsphere/v29.0.0/README.md:7]: See See https://github.com/giantswarm/releases/pull/1459#discussion_r1812133570..

File: vsphere/v29.0.0/release.yaml
Comment (njuettner) [reply vsphere/v29.0.0/release.yaml:98]: We noticed one issue because it wouldn't have a build with os-tooling 1.20.1
```
capv/flatcar-stable-3975.2.1-kube-1.29.9-tooling-1.20.0-gs/flatcar-stable-3975.2.1-kube-v1.29.9.ova
```

Comment (Gacko) [reply vsphere/v29.0.0/release.yaml:98]: If this is the only issue, we can build the image. 🙂

Comment (vxav) [reply vsphere/v29.0.0/release.yaml:98]: I'm lost with all these image versions 😅 Do you need me to upload something?

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_RELEASES=vsphere-29.0.0 PREVIOUS_RELEASE=28.0.1

**Comment by Gacko** [comment]
Take care not to merge this before rebasing after v28.0.1 got merged.

File: vsphere/v29.0.0/README.md
Comment (TheoBrigitte) [inline vsphere/v29.0.0/README.md:15]:
```suggestion
- observability-bundle from v1.5.3 to v1.7.0
```

File: vsphere/v29.0.0/README.md
Comment (TheoBrigitte) [inline vsphere/v29.0.0/README.md:43]:
```suggestion
### observability-bundle [v1.5.3...v1.7.0](https://github.com/giantswarm/observability-bundle/compare/v1.5.3...v1.7.0)
```

File: vsphere/v29.0.0/release.diff
Comment (TheoBrigitte) [inline vsphere/v29.0.0/release.diff:69]:
```suggestion
    version: 1.5.3						|           version: 1.7.0
```

File: vsphere/v29.0.0/release.yaml
Comment (TheoBrigitte) [inline vsphere/v29.0.0/release.yaml:69]:
```suggestion
    version: 1.7.0
```

**Review by TheoBrigitte — COMMENTED** [review]
Can we have observability-bundle `1.7.0` as part of this release ?

**Comment by Gacko** [comment]
@vxav: Tests are failing, because the image hasn't been copied to vSphere, yet. Can you make sure the image for Flatcar 3975.2.2, Kubernetes 1.29.10 and OS Tooling 1.20.1 is present? Thank you!

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
> @vxav: Tests are failing, because the image hasn't been copied to vSphere, yet. Can you make sure the image for Flatcar 3975.2.2, Kubernetes 1.29.10 and OS Tooling 1.20.1 is present? Thank you!

Copied it 👍🏻

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by glitchcrab** [comment]
apologies for hijacking your PR @njuettner, i'm just testing my kube-vip fix build because i [can't use mt PR to do it](https://github.com/giantswarm/cluster-vsphere/pull/299#issuecomment-2441357537). i'll revert my changes afterwards

**Comment by glitchcrab** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by glitchcrab** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv TARGET_RELEASES=vsphere-29.0.0

**Comment by glitchcrab** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
I'm trying to run it one more time to see if that is a flaky issue:

```
 Reason:       object already exists: (rendered manifests contain a resource that already exists. Unable to continue with install: CustomResourceDefinition "podlogs.monitoring.grafana.com" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "grafana-agent"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "kube-system")
```

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by Gacko** [comment]
This particular issue is a flake, sadly. I've seen it on several workload clusters and even management clusters created from already releases Releases.

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by QuentinBisson** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Comment by QuentinBisson** [comment]
Trying the see if my PRs fixed the issues with the tests

**Comment by QuentinBisson** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by QuentinBisson** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by Gacko** [comment]
run releases-test-suites PREVIOUS_RELEASE=28.0.1

**Comment by QuentinBisson** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by QuentinBisson** [comment]
I think I have a fix for upgrades https://github.com/giantswarm/logging-operator/pull/254

**Comment by QuentinBisson** [comment]
We're releasing the fixed logging-operator but we're blocked by push to aliyun

**Comment by Gacko** [comment]
The same `observability-bundle` version used in this PR is also being used in already released CAPA & CAPZ releases. They worked perfectly fine when they got released and we didn't have any issues with tests.

Now it seems like you recently introduced a change to the `logging-operator` which obviously affects `observability-bundle` versions included in already released CAPA & CAPZ releases and now needs to be fixed to make these releases work again.

As a customer I'd expect these releases to be tested and working, so cluster creation shouldn't break out of nowhere.

Can you please elaborate on what has been changed in `logging-operator` and how this affects existing CAPA & CAPZ releases? I'd expect these releases to be stable and immutable and implementing changes in an MC operator which changes the behavior `observability-bundle` in existing releases definitely breaks this contract.

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/standard PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
@giantswarm/team-rocket I think we can finally move on, if you could take another look?

**Comment by Gacko** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by njuettner** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by QuentinBisson** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by QuentinBisson** [comment]
/run releases-test-suites TARGET_SUITES=./providers/capv/upgrade PREVIOUS_RELEASE=28.0.1 TARGET_RELEASES=vsphere-29.0.0

**Comment by Gacko** [comment]
@QuentinBisson or someone else from @giantswarm/team-atlas: Can you please reply to [this](https://github.com/giantswarm/releases/pull/1459#issuecomment-2450152770)? It would be very helpful, even if only for documentation. Also I'd be interested in what has changed between the different runs of Releases Test Suites as I'd prefer to see them reliably fixed instead of having them pass once out of ten. 🙂

**Comment by QuentinBisson** [comment]
@Gacko I'll write something on monday, I wanted to focus on fixing this first

**Comment by Gacko** [comment]
Ok, thank you!

I'll run the standard tests one last time and merge this PR once they pass.

/run releases-test-suites TARGET_SUITES=./providers/capv/standard TARGET_RELEASES=vsphere-29.0.0

**Review by Gacko — APPROVED** [review]

**Comment by QuentinBisson** [comment]
There's currently a fixed branch on gcapeverde so tests should work 🤞🏻

**Comment by QuentinBisson** [comment]
I was going to run them again anyway 😅

**Comment by QuentinBisson** [comment]
> The same observability-bundle version used in this PR is also being used in already released WC releases. They worked perfectly fine when they got released and we didn't have any issues with tests.
> 
> Now it seems like you recently introduced a change to the logging-operator which obviously affects the observability-bundle in existing WC releases and now needs to be fixed to make these releases work again.
> 
> As a customer I'd expect these releases to be tested and working, so cluster creation shouldn't break out of nowhere.
> 
> Can you please elaborate on what has been changed in logging-operator and how this affects existing WC releases? I'd expect these releases to be stable and immutable and implementing changes in an MC operator, which changes the behavior observability-bundle in existing releases, definitely breaks this contract.

We are indeed configuring the observability-platform apps in releases via operators. We initialy built the logging and observability operators as a safety mechanism to be able to change some of our apps config on the fly (for prometheus-agent and so on) to counteract the lack/slowness of customer upgrades in the past because we were swarmed with lots of day and night alerts and that was unsufferable and waiting for a customer to upgrade was a no-go.

We used this mechanism to build some features on our apps as well like:

- sharding of the monitoring agent based on the metrics on prometheus and Mimir (because that would require someone managing KEDA on WCs and no one wants to)
- secret management
- dynamic enabling/disabling of loggging and monitoring at the cluster level

This used to work quite well in the past but we recently enabled a feature flag on the logging-operator that replaced promtail with alloy in the observability platform https://github.com/giantswarm/logging-operator/pull/246 which caused issues last week.

This changed had been manually tested in the past but it missed that the grafana-agent application was failing to deploy (because of an issue with it's CRD management in our currently deployed release of it) which broke the cluster creation test.

In the mean time, a configuration breaking change in the alloy secret management was introduced in the observability-bundle and the change was not properly reflected in the logging-operator which caused the upgrade test to fail because the secret for alloy was not created and so alloy in CAPV 28 did not actually deploy :(

Last week, this became problematic and we are definitely sorry about all of this :(. I am opening a PM today so we can investigate how we can move forward without a lot of the operator work going on behind the hood (we need some config coming from MCs like secret to talk to loki and so on but not as much as we have today) but it should be our priority that we find something that does not break any existing releases.
Would you be up to a discussion to find out how we could integrate better?

By the way, we've been having discussions about this topic for years now and I really thought everyone was aware of it. We really need to find a better way to move forward (cc @JosephSalisbury) and that will require improvements on the release and delivery process as well :)

**Pull request merged**
